### PR TITLE
Add missing parameters in `Mailbox.initLocalValues()`

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/models/mailbox/Mailbox.kt [hunk 1]
```diff
@@ -108,12 +108,25 @@ class Mailbox : RealmObject {
...
+        signatures: List<Signature>?,
+        featureFlags: Set<String>?,
+        externalMailFlagEnabled: Boolean?,
+        trustedDomains: List<String>?,
+    ) {
         this.objectId = createObjectId(userId)
         this.userId = userId
         this.quotas = quotas
-        this.unreadCountLocal = inboxUnreadCount
+        this.unreadCountLocal = inboxUnreadCount ?: 0
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/data/models/mailbox/Mailbox.kt:124]: Shouldn't we use a `let` to only assign the value if it's non null?

**Review by LunarX — CHANGES_REQUESTED** [review]

**Comment by tevincent** [comment]
Your brain is overloaded.

Capture d’écran 2024-09-27 à 10 38 32

**Review by tevincent — CHANGES_REQUESTED** [review]
Brain overloaded 🤯

Capture d’écran 2024-09-27 à 10 38 32

**Comment by tevincent** [comment]
Stop spamming the rebase label @NicolasBourdin88 

Capture d’écran 2024-09-27 à 10 42 03

**Review by tevincent — CHANGES_REQUESTED** [review]
Do we really want to accept that issue ? 😔

Capture d’écran 2024-09-27 à 11 15 19

**Review by LunarX — APPROVED** [review]

**Pull request merged**